daffodil-core/src/test/scala/org/apache/daffodil/general/TestRuntimeProperties.scala

```diff
@@ -0,0 +1,187 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.general
+
+import org.junit.Test
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+
+import org.apache.daffodil.Implicits.intercept
+import org.apache.daffodil.dpath.NodeInfo
+import org.apache.daffodil.infoset.DISimple
+import org.apache.daffodil.infoset.ScalaXMLInfosetInputter
+import org.apache.daffodil.infoset.ScalaXMLInfosetOutputter
+import org.apache.daffodil.io.InputSourceDataInputStream
+import org.apache.daffodil.util.SchemaUtils
+import org.apache.daffodil.util.TestUtils
+
+/**
+ * Test InfosetOutputter that extends the ScalaXMLInfosetOutputter and redacts
+ * words specified in the dfdlx:runtimeProperties extension
+ */
+class RedactingScalaXMLInfosetOutputter
+  extends ScalaXMLInfosetOutputter {
+
+  override def startSimple(diSimple: DISimple): Boolean = {
+    super.startSimple(diSimple)
+
+    val runtimeProperties = diSimple.erd.runtimeProperties
+
+    val redactions = runtimeProperties.get("redact").toSeq.flatMap { key => key.split(",") }
+    if (!redactions.isEmpty) {
+      val thisElem = stack.top.last.asInstanceOf[scala.xml.Elem]
+      var text = thisElem.child(0).asInstanceOf[scala.xml.Text].data
+      val redactWord = runtimeProperties.get("redactWord").getOrElse("")
+      redactions.foreach { redaction =>
+        text = text.replace(redaction, redactWord)
+      }
+      val newChild = Seq(new scala.xml.Text(text))
+      val newElem = thisElem.copy(child = newChild)
+      stack.top(stack.top.size - 1) = newElem
+    }
+    true
+  }
+}
+
+/**
+ * Test InfosetInputter that extends the ScalaXMLInfosetInputter and redacts
+ * words specified in the dfdlx:runtimeProperties extension
+ */
+class RedactingScalaXMLInfosetInputter(rootNode: scala.xml.Node)
+  extends ScalaXMLInfosetInputter(rootNode) {
+
+  override def getSimpleText(primType: NodeInfo.Kind, runtimeProperties: Map[String, String]): String = {
+    var text = super.getSimpleText(primType)
+
+    val redactions = runtimeProperties.get("redact").toSeq.flatMap { key => key.split(",") }
+    if (!redactions.isEmpty) {
+      val redactWord = runtimeProperties.get("redactWord").getOrElse("")
+      redactions.foreach { redaction =>
+        text = text.replace(redaction, redactWord)
+      }
+    }
+    text
+  }
+}
+
+class TestRuntimeProperties {
+
+  val testSchema = SchemaUtils.dfdlTestSchema(
+    <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>,
+    <dfdl:format ref="tns:GeneralFormat" lengthKind="delimited"/>,
+    <xs:element name="root">
+      <xs:complexType>
+        <xs:sequence dfdl:separator=",">
+          <xs:element name="filtered" type="xs:string" dfdlx:runtimeProperties="redact=foo,baz redactWord=REDACTED" />
```

Comment:
Per suggestion above, make another test schema which is just like this one, but which adds a pattern facet to filtered disallowing the "REDACTED" from showing up in the data. So that if redaction occurs, the resulting data will be invalid. Then have tests detect this validation error, but only if "full" validation is on.